datadog/resource_datadog_monitor.go

```diff
@@ -378,7 +396,14 @@ func buildMonitorStruct(d builtResource) (*datadogV1.Monitor, *datadogV1.Monitor
 	if attr, ok := d.GetOk("evaluation_delay"); ok {
 		o.SetEvaluationDelay(int64(attr.(int)))
 	}
-	if attr, ok := d.GetOk("no_data_timeframe"); ok {
+	attr, onMissingDataOk := d.GetOk("on_missing_data")
+	if onMissingDataOk {
+		o.SetOnMissingData(datadogV1.OnMissingDataOption(attr.(string)))
+	}
+	// no_data_timeframe cannot be combined with on_missing_data. This provider
+	// defaults no_data_timeframe to 10, so we need this extra logic to exclude
+	// no_data_timeframe from the monitor definition when on_missing_data is set.
+	if attr, ok := d.GetOk("no_data_timeframe"); ok && !onMissingDataOk {
```

Comment:
👍 this will preferentially set on_missing_data if it's set, which is what we want